api/src/test/java/org/openmrs/ConceptTest.java

```diff
@@ -570,6 +581,65 @@ public void getSetMembers_shouldReturnUnmodifiableListOfConceptMemberList() {
 		Assert.assertEquals(1, setMembers.size());
 		setMembers.add(new Concept());
 	}
+
+	/**
+	 * @see Concept#getSetMembers()
+	 */
+	@Test
+	public void getSetMembers_shouldReturnTheSameAsgetSetMembersIfincludeRetiredIsTrue() throws Exception {
+		executeDataSet(CONCEPT_XML_DATASET_PACKAGE_PATH);
+
+		Concept c = new Concept();
+
+		Concept setMember1 = service.getConcept(867543);
+		c.addSetMember(setMember1);
+
+		Concept setMember2 = service.getConcept(1234567);
+		c.addSetMember(setMember2);
+
+		Concept setMember3 = service.getConcept(8675439);
+		c.addSetMember(setMember3);
+
+		Concept setMember4 = service.getConcept(12345679);
+		c.addSetMember(setMember4);
+
+
+		List<Concept> setMembers = c.getSetMembers(true);
+
+		Assert.assertEquals(4, setMembers.size());
+		Assert.assertEquals(setMember1, setMembers.get(0));
+		Assert.assertEquals(setMember2, setMembers.get(1));
+	}
+
+	/**
+	 * @see Concept#getSetMembers(boolean)
+	 */
+	@Test
+	public void getSetMembers_shouldNotReturnRetiredInSetMembersIfincludeRetiredIsFalse() throws Exception {
+		executeDataSet(CONCEPT_XML_DATASET_PACKAGE_PATH);
+
+		Concept c = new Concept();
+
+		Concept setMember1 = service.getConcept(867543);
+		c.addSetMember(setMember1);
+
+		Concept setMember2 = service.getConcept(1234567);
+		c.addSetMember(setMember2);
+
+		Concept setMember3 = service.getConcept(8675439);
+		c.addSetMember(setMember3);
+
+		Concept setMember4 = service.getConcept(12345679);
+		c.addSetMember(setMember4);
+
+		List<Concept> setMembers = c.getSetMembers(false);
+
+		Assert.assertEquals(2, setMembers.size());
```

Comment:
How can you confirm that these two objects are unretired objects and no more retired objects in this list?

Reply:
setMembers.forEach(member -> Assert.assertEquals(true, !member.getRetired()));
this will go through each member and check if they are not retired

Reply:
Let's do it